lib/hanami/view/application_configuration.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative "../view"
+
+module Hanami
+  class View
+    class ApplicationConfiguration
+      def initialize
+        @base_config = View.config.dup
+
+        configure_defaults
+      end
+
+      # Returns the list of available settings
+      #
+      # @return [Set]
+      #
+      # @since 2.0.0
+      # @api private
+      def settings
+        View.settings - NON_FORWARDABLE_METHODS
+      end
+
+      private
+
+      attr_reader :base_config
+
+      def configure_defaults
+        self.paths = ["web/templates"]
+        self.template_inference_base = "views"
+        self.layout = "application"
+      end
+
+      # An inflector for views is not configurable via `config.views.inflector` on an
+      # `Hanami::Application`. The application-wide inflector is already configurable
+      # there as `config.inflector` and will be used as the default inflector for views.
+      #
+      # A custom inflector may still be provided in an `Hanami::View` subclass, via
+      # `config.inflector=`.
+      NON_FORWARDABLE_METHODS = [:inflector, :inflector=].freeze
+      private_constant :NON_FORWARDABLE_METHODS
+
+      def method_missing(name, *args, &block)
+        return super if NON_FORWARDABLE_METHODS.include?(name)
```

Comment:
@timriley Probably this `return` is redundant because calling `super` will raise a `NoMethodError` exception, which is what we're looking for.

Reply:
@jodosha Do you mean the `super` in the else branch? As far as I see it, we still need this guard clause up top because `base_config` _does_ respond to `inflector` and `inflector=`, but in this case we do _not_ want to forward those methods, because we're hiding them in this context.

Does that make sense?

Reply:
Since you've ticked this PR, I'm going to merge this now anyway, but happy to tweak this particular bit after further discussion :)